ssl/t1_lib.c

```diff
@@ -997,6 +997,24 @@ static const SIGALG_LOOKUP *tls1_get_legacy_sigalg(const SSL *s, int idx)
                     }
                 }
             }
+            /*
+             * Here is another fallback: when broken implementations did not sent
+             * proper signature_algorithm extension, we try to use this function.
+             *
+             * As both SSL_PKEY_GOST12_512 and SSL_PKEY_GOST12_256 indices can be used
+             * with new (aGOST12-only) ciphersuites, we should find out which one is available really.
```

Comment:
Are there broken implementations that support these sigalgs but don't send the sigalgs extension?

Reply:
Unfortunately, yes. As the developers of such implementations (mostly Windows-based) has explained, there is a lack of the corresponding hook in Windows to alter the list of SigAlgs.